sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/thriftserver/CliSuite.scala

```diff
@@ -500,4 +500,13 @@ class CliSuite extends SparkFunSuite with BeforeAndAfterAll with BeforeAndAfterE
         |   ;""".stripMargin -> "testcomment"
     )
   }
+
+  test("SPARK-31595 Should allow unescaped quote mark in quoted string") {
+    runCliWithin(1.minute)(
```

Comment:
Could you explicitly set false at `spark.sql.parser.escapedStringLiterals` for the tests below?

Reply:
Actually this has nothing to do with `spark.sql.parser.escapedStringLiterals`, even if the config is set to true, the parser should accept this string.

Reply:
Ur, I got it. I think the option is misleading... Could you remove it in the PR descritpion?

Reply:
Done that, thanks!